crates/dbsp/src/utils/sample.rs

```diff
@@ -1,34 +1,24 @@
-use crate::dynamic::LeanVec;
 use rand::{seq::index::sample, RngCore};
 
 /// Compute a random sample of size `sample_size` of values in `slice`.
 ///
-/// Pushes the random sample of values to the `output` vector in the order
-/// in which values occur in `slice`, i.e., if `slice` is sorted, the output
-/// will be sorted too.
-pub fn sample_slice<T>(
-    slice: &[T],
-    rng: &mut dyn RngCore,
-    sample_size: usize,
-    output: &mut LeanVec<T>,
-) where
-    T: Clone,
+/// Invokes `callback` for each value in the sample in the order
+/// in which values occur in `slice`.
+pub fn sample_slice<T, F>(slice: &[T], rng: &mut dyn RngCore, sample_size: usize, callback: &mut F)
```

Comment:
It would be cool if this returned an iterator instead of taking a callback.

But it's fine as-is.